pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -117,25 +117,108 @@ func (p *PoolManager) ReleaseVirtualMachineMac(virtualMachineName string) error
 	return nil
 }
 
-func (p *PoolManager) allocateFromPoolForVirtualMachine(virtualMachine *kubevirt.VirtualMachine) (string, error) {
+func (p *PoolManager) UpdateMacAddressesForVirtualMachine(virtualMachine *kubevirt.VirtualMachine) error {
+	p.poolMutex.Lock()
+
+	log.V(1).Info("UpdateMacAddressesForVirtualMachine: data",
+		"macmap", p.macPoolMap,
+		"podmap", p.podToMacPoolMap,
+		"vmmap", p.vmToMacPoolMap,
+		"currentMac", p.currentMac.String())
+
+	existInterfacesMap, isVirtualMachineExist := p.vmToMacPoolMap[vmNamespaced(virtualMachine)]
+	if !isVirtualMachineExist {
+		p.poolMutex.Unlock()
```

Comment:
We can remove this unlock moving the defer before the if statement.

Reply:
I think it will be a deadlock because the `p.AllocateVirtualMachineMac(virtualMachine)` try to lock the mutex.

correct me but the defer calls after the function finish no if I call another function right?

Reply:
If this is the case then this is alright.